src/js/api/config.js

```diff
@@ -61,9 +61,53 @@ function _adjustDefaultBwEstimate(estimate) {
     return Defaults.bandwidthEstimate;
 }
 
+function _mergeProperty(localizationObj, allOptionsObj, prop) {
+    if (prop) {
+        const propToCopy = localizationObj[prop] || allOptionsObj[prop];
+
+        if (propToCopy) {
+            localizationObj[prop] = propToCopy;
+        }
+    } else {
+        localizationObj = Object.assign({}, localizationObj, allOptionsObj);
+    }
+}
+
+function _copyToLocalization(allOptions) {
+    const { advertising, related, sharing, abouttext } = allOptions;
+    const localization = Object.assign({}, allOptions.localization);
+
+    if (advertising) {
+        _mergeProperty(localization.advertising, advertising);
```

Comment:
let's use object.assign here and remove the `else` block from `_mergeProperty` since that function should be property specific.
Also, I made a mistake when I told you to use the entire advertising block in `Object.assign({}, localization.advertising, advertising)` instead of your own object containing the specific advertising options. We should revert it to what you had before since the advertising block will contain other fields we don't want